cs/src/core/Device/ManagedLocalStorageDevice.cs

```diff
@@ -48,14 +51,19 @@ private void RecoverFiles()
 
             string bareName = fi.Name;
 
-            int prevSegmentId = -1;
+            List<int> segids = new List<int>();
```

Comment:
Maybe I am missing something, but it's unclear to me why the list is required. Is it just for readability?

Reply:
The list of files comes in alphabetical order by default, not numerical. Caused a bug without sorting.